comp/netflow/config/mock.go

```diff
@@ -0,0 +1,30 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package config
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"go.uber.org/fx"
+)
+
+func newMock(conf *NetflowConfig) Component {
+	return &configService{conf}
+}
+
+// MockModule defines the fx options for the mock component.
+// Injecting MockModule will provide default config;
+// override this with fx.Replace(&config.NetflowConfig{...}).
+// Note that defaults will always be populated.
+var MockModule = fxutil.Component(
+	fx.Provide(newMock),
+	fx.Supply(&NetflowConfig{}),
+	fx.Decorate(
```

Comment:
`fx.Decorate` is not needed as the user can use `fx.Replace` in unit tests to replace the instance of `NetflowConfig{}`.

Reply:
I will remove this and just call SetDefaults() in the test.

Reply:
The `fx.Decorate` is actually making the tests simpler - it gets called on replacement values, so we can just write
```go
s := fxutil.Test[Component](t, ..., fx.Replace(&nfconfig.NetflowConfig{...}))
```
and the replacement netflow config will automatically have `SetDefaults()` called. 

Without the `Decorate`,  the test has to call `SetDefaults` itself:
```
conf := &nfconfig.NetflowConfig{...}
if err := conf.SetDefaults(); err != nil {
    t.Fatal(err)
}
s := fxutil.Test[Component](t, ..., fx.Replace(conf))
```

Reply:
Is there a drawback of calling `conf.SetDefaults("default")` inside `newMock`?

Reply:
No, it just didn't occur to me for some reason. Fixed.